### PR TITLE
Add Chart widget

### PR DESCRIPTION
**Pull request by AIDDQD**

### CODE REVIEW

File: chart.go [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+}
+
+func (c *Chart) Clear() {
+	C.go_fltk_Chart_clear((*C.GChart)(c.ptr))
+}
+
+//Add the data value val with optional label text and color col to the chart.
+//When color not needed just pass zero
+func (c *Chart) Add(val float32, col Color, text ...string) {
+	//TODO: Should i use float32 or better replace it with float64?
```
Comment (pwiecz) [inline chart.go:36]: I'd use float64 to use types that are equivalent to the ones used by FLTK. (same below)

File: chart.go [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+	C.go_fltk_Chart_add((*C.GChart)(c.ptr), C.double(val), cStringOpt(text), C.uint(col))
```
Comment (pwiecz) [inline chart.go:37]: Have you checked that the Chart takes ownership of the passed char* ?
Otherwise we have a memory leak here.

**Review by pwiecz — COMMENTED** [review]
Please look at comments. :)

File: chart.go (hunk 2 above)
Comment (AIDDQD) [reply chart.go:37]: I will check it. If chart doesn't should i call C.free on cStringOpt return value?

Comment (pwiecz) [reply chart.go:37]: Yes, similarly as in: https://github.com/pwiecz/go-fltk/blob/main/widget.go#L94

Comment (AIDDQD) [reply chart.go:37]: image
Seems like it just copy string to interal array

Comment (AIDDQD) [reply chart.go:37]: So, I'll just fix this

**Comment by AIDDQD** [comment]
Oh, forgot to test it before commit. Don't approve pull request

**Comment by pwiecz** [comment]
Please tell me once you think it's safe to merge. :)

**Comment by AIDDQD** [comment]
I plan to add an example of the chart widget usage, should i open new pull request?

**Comment by pwiecz** [comment]
I you wish, I can wait for you to add another commit here.
It seems to be a good fit for this pull request.

**Pull request merged**